### PR TITLE
tests: Return error if listener fails to close after checking if port is free

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

This returns an error if the listener fails to close after checking if the port is free
This was discussed during a huddle with @sam-heilbron and @sheidkamp 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: test/ginkgo/parallel/ports.go [hunk 1]
```diff
@@ -25,15 +25,23 @@ func GetPortOffset() int {
 	return GetParallelProcessCount() * 1000
 }
 
+// RetryWithLoggedError retries if there is no error and prints any error that occurs to stdout
```
Comment (sam-heilbron) [inline test/ginkgo/parallel/ports.go:28]: nit: retries if there "is an error"

File: test/ginkgo/parallel/ports.go [hunk 2]
```diff
@@ -77,7 +85,6 @@ func portInUseListen(proposedPort uint32) error {
 		return eris.Wrapf(err, "port %d is in use", proposedPort)
 	}
 
-	_ = ln.Close()
 	// Port is available
```
Comment (sam-heilbron) [inline test/ginkgo/parallel/ports.go:88]: nit: The port is available comment may not be entirely accurate

**Review by sam-heilbron — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

File: changelog/v1.18.0-beta13/handle-failed-closed-listener.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changelog:
+- type: NON_USER_FACING
+  description: Return the error if it fails to close the listener after testing if the port is free.
```
Comment (nfuden) [inline changelog/v1.18.0-beta13/handle-failed-closed-listener.yaml:3]:
```suggestion
  description: Print error in tests if tests fail to close net.Listen after testing if a port is free.
```

**Review by nfuden — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Comment by sam-heilbron** [comment]
I have a PR https://github.com/solo-io/gloo/pull/9844 which will require some fast-follow edits. It's scheduled to enter CI that same time as this PR. To try to coordinate merges, I've add a work in progress label. My plan is:
- Let https://github.com/solo-io/gloo/pull/9844 merge
- I can begin a PR that relies on those changes
- Remove the work in progress label on this PR
- Let this merge

**Pull request merged**